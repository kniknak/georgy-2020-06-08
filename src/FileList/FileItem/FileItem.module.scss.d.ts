declare const styles: {
    readonly root: string;
    readonly filename: string;
    readonly size: string;
    readonly button: string;
};
export = styles;