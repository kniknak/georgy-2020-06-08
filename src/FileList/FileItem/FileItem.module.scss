@mixin mobile {
    @media only screen and (max-width: 375px) {
        @content;
    }
}

@mixin tablet {
    @media only screen and (max-width: 600px) {
        @content;
    }
}

.root {
    min-width: calc(33.3% - 12px);
    max-width: calc(33.3% - 12px);
    box-sizing: border-box;
    border: 1px solid cornflowerblue;
    border-radius: 4px;
    padding: 12px;
    margin: 6px;
    display: flex;
    flex-wrap: wrap;

    @include tablet {
        min-width: calc(50% - 12px);
        max-width: calc(50% - 12px);
    }

    @include mobile {
        min-width: calc(100% - 12px);
        max-width: calc(100% - 12px);
    }
}

.filename {
    width: 100%;
    overflow: hidden;
    white-space: nowrap;
    text-overflow: ellipsis;
    font-size: 24px;
    margin-bottom: 24px;
}

.size {
    white-space: nowrap;
}

.button {
    margin-left: auto;
    background-color: transparent;
    border: 1px solid cornflowerblue;
    border-radius: 4px;
    padding: 4px 12px;
    cursor: pointer;
}
