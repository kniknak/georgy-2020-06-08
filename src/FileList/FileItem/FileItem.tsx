import React, { useCallback } from "react";
import { FileInfo } from "../../api/types";
import * as styles from "./FileItem.module.scss";
import { formatSize } from "../formatSize";

export interface FileItemProps {
    file: FileInfo
    deleteFile: (id: string) => void
}

export const FileItem: React.FC<FileItemProps> = ({ file, deleteFile }) => {
    const onClick = useCallback(() => deleteFile(file._id), [deleteFile, file]);

    return (
        <li className={styles.root}>
            <div className={styles.filename}>{file.originalName}</div>
            <div className={styles.size}>{formatSize(file.size)}</div>
            <button
                className={styles.button}
                onClick={onClick}
            >
                Delete
            </button>
        </li>
    );
};
