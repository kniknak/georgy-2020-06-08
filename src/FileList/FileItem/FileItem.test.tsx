import React from "react";
import { shallow } from "enzyme";
import { FileItem, FileItemProps } from "./FileItem";

describe("FileList", () => {
    describe("FileItem", () => {
        const defaultProps: FileItemProps = {
            file: {
                originalName: "foo",
                size: 42,
                _id: "bar",
                createdAt: 200,
            },
            deleteFile: () => undefined,
        };

        it("renders", () => {
            const subject = shallow(<FileItem {...defaultProps}/>);

            expect(subject).toMatchSnapshot();
        });

        it("handles file delete", () => {
            const props: FileItemProps = {
                ...defaultProps,
                file: {
                    ...defaultProps.file,
                    _id: "fileId",
                },
                deleteFile: jest.fn(),
            };
            const subject = shallow(<FileItem {...props}/>);

            subject.find("button").prop("onClick")!({} as any);
            expect(props.deleteFile).toHaveBeenLastCalledWith(props.file._id);
        });
    });
});