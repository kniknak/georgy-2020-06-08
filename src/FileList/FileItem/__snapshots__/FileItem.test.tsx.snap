// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileList FileItem renders 1`] = `
<li>
  <div>
    foo
  </div>
  <div>
    42b
  </div>
  <button
    onClick={[Function]}
  >
    Delete
  </button>
</li>
`;
